tests/installation/addon_products_yast2.pm

```diff
@@ -75,44 +75,49 @@ sub run {
             send_key 'alt-a';                                                       # accept
             assert_screen 'automatic-changes';
             send_key 'alt-o';                                                       # OK
-            if (check_screen 'unsupported-packages', 5) {
-                record_soft_failure 'unsupported packages';
-                send_key 'alt-o';
-            }
-            if (check_screen 'addon-installation-pop-up', 100) {                    # e.g. RT reboot to activate new kernel
-                $perform_reboot = 1;
-                send_key 'alt-o';                                                   # OK
-            }
-            assert_screen "addon-installation-report";
-            send_key 'alt-f';                                                       # finish
+            my @needles = qw(unsupported-package addon-installation-pop-up addon-installation-report);
+            do {
+                assert_screen \@needles, 300;
+                if (match_has_tag('unsupported-packages')) {
+                    record_soft_failure 'unsuppoorted packages';
+                    send_key 'alt-o';
+                }
+                if (match_has_tag('addon-installation-pop-up')) {
+                    $perform_reboot = 1;
+                    send_key 'alt-o';
+                }
+            } until (match_has_tag('addon-installation-report'));
+            wait_screen_change { send_key 'alt-f'; }    # finish
+            send_key 'alt-a';                           # accept
+            sleep 10;
```

Comment:
I told you already in before: no sleeps or check_screen with non-zero timeout

Reply:
Oops, thats a relict